File: backend/springboot/src/main/java/tn/maiko26/springboot/models/relations/UserNotification.java
package tn.maiko26.springboot.models.relations;

import jakarta.persistence.*;
import tn.maiko26.springboot.models.Notification;
import tn.maiko26.springboot.models.User;

import java.io.Serializable;

@Entity
@Table(name = "user_notifications")
@IdClass(UserNotificationId.class)
public class UserNotification {
    @Id
    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "email")
    private User user;

    @Id
    @ManyToOne
    @JoinColumn(name = "notification_id")
    private Notification notification;
}

 class UserNotificationId implements Serializable {
    private String user;
    private String notification;
}

File: backend/dotnet/Models/relations/UserNotification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace dotnet.Models;

[Table("user_notifications")]
public class UserNotification
{
    [Key, Column("user_id")]
    public string UserId { get; set; } = string.Empty;

    [Key, Column("notification_id")]
    public string NotificationId { get; set; } = string.Empty;

    // Navigation properties
    [JsonIgnore]
    public User User { get; set; } = null!;
    [JsonIgnore]
    public Notification Notification { get; set; } = null!;
}

